### PR TITLE
Disable snapping when the add feature tool is activated on a non-spatial vector layer

### LINKED ISSUE #56152: Add a new record in a table enable the snapping tool
### What is the bug or the crash?

After adding a new record in a table, the snapping tool is enabled

### Steps to reproduce the issue

1. Insert a table layer (without geometry)
2. Activate the edit mode for the layer
3. Turn on pan navigation (to highlight the existing issue)
4. Click on "Add a new record"
5. Confirm or close the form
6. Observe that the snapping tool becomes active (the cursor persists as the pan tool, even though panning functionality is not enabled)

*(The snapping square displayed in the animation should be purple instead of white)*

snapping

### Versions

3.28.13

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by benwirf**
## Description

This PR fixes a reported bug whereby active snapping settings are enabled when adding records to a non-spatial table.

Some logic is also added to manage setting the correct digitizing cursors when switching between spatial and non-spatial vector layer with both in edit mode and add feature tool active.

Fixes #56152

Screenshot showing current behavior (snapping enabled *also note that the open hand cursor persists when switching to a spatial layer in edit mode):

snapping-bug-1

Screenshot showing snapping now disabled when adding records to non-spatial layer.

snapping-bug-2

And with advanced digitizing enabled (also showing better cursor handling)

snapping-bug-3

### CODE REVIEW

**Comment by benwirf** [comment]
Ping @nirvn

File: src/gui/maptools/qgsmaptooladvanceddigitizing.cpp [hunk 1]
```diff
@@ -82,6 +90,10 @@ void QgsMapToolAdvancedDigitizing::canvasReleaseEvent( QgsMapMouseEvent *e )
 
 void QgsMapToolAdvancedDigitizing::canvasMoveEvent( QgsMapMouseEvent *e )
 {
+  if ( canvas()->currentLayer() && !canvas()->currentLayer()->isSpatial() )
+  {
+    return;
+  }
```
Comment (nirvn) [inline src/gui/maptools/qgsmaptooladvanceddigitizing.cpp:96]: Why are these blocks needed? You're already disabling the cad dock widget in the onCurrentLayerChanged() function.

Comment (benwirf) [reply src/gui/maptools/qgsmaptooladvanceddigitizing.cpp]: @nirvn, my bad. Yes you're right- they're not needed. In my older PR, I wasn't disabling the cad dock widget (only clearing it). I added the `disable()` call here which I think is the right approach, but forgot to remove those checks.
I have just finished my day job so I've now removed them & updated the branch.
Thanks for looking at this- it's much appreciated.

**Review by nirvn — APPROVED** [review]

File: src/gui/maptools/qgsmaptoolcapture.cpp [hunk 2]
```diff
@@ -154,6 +154,16 @@ void QgsMapToolCapture::currentLayerChanged( QgsMapLayer *layer )
     return;
   }
 
+  if ( vlayer->isSpatial() )
```
Comment (3nids) [inline src/gui/maptools/qgsmaptoolcapture.cpp:157]: This is not the correct approach.
This map tool is a base class and is used to capture even if there is no current layer.
This is breaking this scenerio.

File: src/gui/maptools/qgsmaptoolcapture.cpp [hunk 3]
```diff
@@ -476,8 +486,8 @@ void QgsMapToolCapture::cadCanvasMoveEvent( QgsMapMouseEvent *e )
   QgsMapToolAdvancedDigitizing::cadCanvasMoveEvent( e );
 
   const QgsPointXY point = e->mapPoint();
-
-  mSnapIndicator->setMatch( e->mapPointMatch() );
+  if ( canvas()->currentLayer() && canvas()->currentLayer()->isSpatial() )
```
Comment (3nids) [inline src/gui/maptools/qgsmaptoolcapture.cpp:489]: This is not the correct approach.
This map tool is a base class and is used to capture even if there is no current layer.
This is breaking this scenerio.

Comment (benwirf) [reply src/gui/maptools/qgsmaptoolcapture.cpp:489]: @3nids Oops, you're right! Sorry about that. I have proposed a fix here #60166 which I think resolves the issue if you wouldn't mind taking a look.

**Pull request merged**